### PR TITLE
Migration to ruff

### PR DESCRIPTION
**Pull request by MasterSkepticista**
This PR migrates all lint/format tooling to [ruff](https://github.com/astral-sh/ruff), a modern, fast, rust-based formatter with significantly more capabilities.

Ruff has a drop-in parity with flake8, black and isort - all linting tools used in OpenFL. All config now resides in `pyproject.toml`.

### CODE REVIEW

File: openfl/experimental/workflow/component/aggregator/aggregator.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def call_checkpoint(self, ctx: Any, f: Callable, stream_buffer: bytes = None) ->
                 f = pickle.loads(f)
             if isinstance(stream_buffer, bytes):
                 # Set stream buffer as function parameter
-                setattr(f.__func__, "_stream_buffer", pickle.loads(stream_buffer))
+                f.__func__._stream_buffer = pickle.loads(stream_buffer)
```
Comment (MasterSkepticista) [inline openfl/experimental/workflow/component/aggregator/aggregator.py:236]: This is an example of simplification automatically made by ruff.

**Review by tanwarsh — APPROVED** [review]
LGTM.

**Comment by tanwarsh** [comment]
@MasterSkepticista Can you please add changes for Check code format pipeline as well.

**Comment by MasterSkepticista** [comment]
> @MasterSkepticista Can you please add changes for Check code format pipeline as well.

It is updated, I'm refactoring warnings raised by ruff, few of the remaining ones are mentioned in #1177. Would appreciate any help!

**Pull request merged**